docs/ru/data_types/float.md

```diff
@@ -1,7 +1,60 @@
-Float32, Float64
-----------------
+# Float32, Float64
 
-Числа с плавающей запятой, то же, что и float, double в языке C.
-В отличие от стандартного SQL, числа с плавающей запятой поддерживают inf, -inf, а также nan-ы.
-Смотрите замечание о сортировке nan-ов в разделе "Секция ORDER BY".
-Не рекомендуется хранить числа с плавающей запятой в таблицах.
+[Числа с плавающей точкой](https://en.wikipedia.org/wiki/IEEE_754).
+
+Типы эквивалентны типам языка С:
+- `Float32` - `float`;
+- `Float64` - `double`.
+
+Из соображений производительности рекомендуем хранить данные в целочисленноми виде всегда, когда это возможно. Например, переводите в целочисленные значения числа с фиксированной точностью, такие как денежные суммы или времена загрузки страниц в миллисекундах.
```

Comment:
Performance is not the reason.

Reply:
There is no real problems of storing floats in tables.
We are doing it in many places...
Example: we store data of Graphite (monitoring system) in ClickHouse, and value of metric is stored in column of type Float64. It works fine for that purpose.

But there are some considerations:

1. Using floats for money is discouraged.

2. When there is value with fixed precision (for example, page loading time in milliseconds), it's much easier and efficient to store it as integer (as number of milliseconds in this example).

3. Parsing of floats from text formats is inexact: not closest machine representable number is parsed, but with some little rounding error.

4. Comparing results of different calculations in floats are more difficult. Example: SELECT 1 - 0.9 = 0.1 returns 0.

5. Floats are more difficult by existence of infs, nans, negative zero. Note that in standard SQL these values are prohibited in FLOAT, REAL data types, but in ClickHouse, Float64 and Float32 are machine floats.

6. Result of calculation may depend on calculation method (what kind of CPU registers and instructions was used), that is implementation specific.